src/js/literal-map.js

```diff
@@ -0,0 +1,55 @@
+const { get, getOwnPropertyDescriptor, ownKeys } = Reflect;
+
+const getPropertyDescriptor = (value) => ({
+  value,
+  enumerable: true,
+  writable: true,
+  configurable: true,
+});
+
+const _ = Symbol();
+const prototype = "prototype";
+
+const handler = {
+  deleteProperty: (map, k) => (map.has(k) ? map.delete(k) : delete map[k]),
+  get(map, k, proxy) {
+    if (k === _) return map;
+    let v = map[k];
+    if (typeof v === "function" && k !== "constructor") {
+      v = v.bind(map);
+    }
+    v ||= map.get(k);
+    return v;
+  },
+  getOwnPropertyDescriptor(map, k) {
+    if (map.has(k)) return getPropertyDescriptor(map.get(k));
+    if (k in map) return getOwnPropertyDescriptor(map, k);
+  },
+  has: (map, k) => map.has(k) || k in map,
+  ownKeys: (map) =>
+    [...map.keys(), ...ownKeys(map)].filter((x) =>
+      ["string", "symbol"].includes(typeof x),
```

Comment:
looks like this is a good thing to do but I wonder why this file is mostly a copy/paste of the library as opposite of being the library. It's harder for me to track possible discrepancies if this gets adopted by other interpreters too.

As this and the `constructor` check seems to be the only difference, and both make sense to me, would you consider using the library if I reflect those changes in there? 🤔

Reply:
This filter is needed because the `ownKeys` trap raises if the list returned contains anything that is not a string or symbol so

Reply:
Makes sense, I’ve mistaken the key with its value, although is not clear to me when a literal key wouldn’t be a string or symbol but I see this playing a role in the dictionary key case.

would you agree this should rather be upstream in the source lib? 🤔

I will implement this ASAP, with or without your consent 😉

Reply:
FYI this change (only this one + the constructor one as I have concerns about the retro-shadowing) landed and I think it should now work out of the box without surprises.